models/minitia/tx.go

```diff
@@ -345,3 +349,149 @@ const FundMinitiaAccountsWithoutBatchTxInterface = `
   "signatures":[]
 }
 `
+
+func (lsk *L1SystemKeys) calculateTotalWantedCoins(state *LaunchState) (l1Want int64, daWant int64, err error) {
+	for _, acc := range []*types.GenesisAccount{
+		lsk.BridgeExecutor,
+		lsk.OutputSubmitter,
+		lsk.BatchSubmitter,
+		lsk.Challenger,
+	} {
+		if acc == nil {
+			continue
+		}
+
+		amount, err := strconv.ParseInt(acc.Coins, 10, 64)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to parse coin amount %s: %v", acc.Coins, err)
+		}
+
+		if acc == lsk.Challenger && state.batchSubmissionIsCelestia {
+			daWant += amount
+		} else {
+			l1Want += amount
+		}
+	}
+
+	return l1Want, daWant, nil
+}
+
+func queryChainBalance(binaryPath, rpc, address string) (map[string]int64, error) {
+	queryCmd := exec.Command(binaryPath, "query", "bank", "balances", address,
+		"--node", rpc, "--output", "json")
+	balanceRes, err := queryCmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to query balance: %v", err)
+	}
+
+	var balance struct {
+		Balances []struct {
+			Denom  string `json:"denom"`
+			Amount string `json:"amount"`
+		} `json:"balances"`
+	}
+	if err := json.Unmarshal(balanceRes, &balance); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal balance: %v", err)
+	}
+
+	balanceMap := make(map[string]int64)
+	for _, bal := range balance.Balances {
+		amount, err := strconv.ParseInt(bal.Amount, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse balance amount: %v", err)
+		}
+		balanceMap[bal.Denom] = amount
+	}
+
+	return balanceMap, nil
+}
+
+func recoverAndCleanupKey(binaryPath, keyName, mnemonic string) (*cosmosutils.KeyInfo, func(), error) {
```

Comment:
we already have cosmosutils.GetAddressFromMnemonic maybe we do not need to implement new function if the purpose is to recover address from mnemonic with binary